### PR TITLE
ENH: Filling undefined data points

### PR DESCRIPTION
**Pull request by pastewka**
This PR implements data imputation, filling of undefined data points with interpolated quantities. The implementation separates the undefined data points into patches and interpolates the values within each patch using harmonic functions. The harmonic function is constructed by solving the Laplace equation with the values on the edge of the patch as Dirichlet boundary conditions. This construction leads to exact imputation for linear fields and has the nice property that the mean of the interpolated field is equal to the mean of the height over the edge (mean value property of harmonic functions). This leads to well behaved interpolations even if large patches are missing.

### CODE REVIEW

File: SurfaceTopography/Uniform/Imputation.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+from ..UniformLineScanAndTopography import DecoratedUniformTopography
+
+# Stencils for determining nearest-neighbor relationships on a square grid
+nn_stencil = [(1, 0), (0, 1), (-1, 0), (0, -1)]
+nnn_stencil = [(1, 1), (1, 0), (1, -1), (0, -1), (-1, -1), (-1, 0), (-1, 1), (0, 1)]
+
+
+def coordination(c, periodic, stencil=nn_stencil):
+    """
+    Return a map with coordination numbers, i.e. number of neighboring patches that also contact
```
Comment (AntoineSIMTEK) [inline SurfaceTopography/Uniform/Imputation.py:44]: Docstring: is this an old code from contactmechanics ? Maybe be more explicit and say the coordination number is computed from the number of neighboring  patches where  c is true

**Review by AntoineSIMTEK — APPROVED** [review]
I didn't try to understand the details of the harmonic interpolation. 

It seems that there is no direct test of the interpoliation. Would it make sense to 1. create a topography that satisfies the laplace equation, 2. remove datapoints, 3. test that the imputation regenerates the correct values ?

File: SurfaceTopography/Uniform/Imputation.py (hunk 1 above)
Comment (pastewka) [reply SurfaceTopography/Uniform/Imputation.py]: Yes this was contact mechanics specific. I've updated the documentation string.

**Comment by pastewka** [comment]
> It seems that there is no direct test of the interpoliation. Would it make sense to 1. create a topography that satisfies the laplace equation, 2. remove datapoints, 3. test that the imputation regenerates the correct values ?

I've added such a test, see `test_laplace_2d`. It's a very simple solution of the laplace equation but shows that the imputation then exactly reproduces the original topography.

**Review by AntoineSIMTEK — APPROVED** [review]

**Pull request merged**